sql-cli/noxfile.py

```diff
@@ -21,21 +21,35 @@ def dev(session: nox.Session) -> None:
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-def test(session: nox.Session) -> None:
+@nox.parametrize("airflow", ["2.1", "2.2.0", "2.3", "2.4"])
+def test(session: nox.Session, airflow: str) -> None:
     """Run both unit and integration tests."""
     session.install("poetry")
     session.run("poetry", "install", "--with", "dev")
-    # Log all the installed dependencies
+    session.run("poetry", "add", f"apache-airflow=={airflow}")
     session.log("Installed Dependencies:")
-    session.run("pip3", "freeze")
-    airflow_home = f"~/airflow-latest-{session.python}"
+    session.run("poetry", "run", "pip3", "freeze")
+
+    # TODO: at the moment flow run depends on the Airflow global Airflow Home - we need to fix this
+    # Refactor so each test does this in their own sandboxed Airflow home
+    # From what I observed, the `run` command is the one using this, due to how the method
+    # utils.airflow.get_dag behaves under Airflow 2.2.0 (it defaults to $HOME/airflow)
+    airflow_home = f"~/airflow-latest-{session.python}-airflow-{airflow}"
```

Comment:
```suggestion
    airflow_home = f"~/airflow-{airflow}-{session.python}"
```
This was meant to be the placeholder for the airflow version 😅

Reply:
@tatiana WDYT of the following?
```suggestion
    airflow_home = f"~/python-{session.python}-airflow-{airflow}"
```

Reply:
Maybe we don't even need the python version, I am not 100% sure. Most errors we get if reuse the airflow home is due to the airflow database right? not due to the python version.

Reply:
I think it is useful for us to have the ability to run concurrently the tests for multiple versions of Python - this protects us from possible issues until we solve the underlining problem of relying on this `$AIRFLOW_HOME`.